dask_cuda/proxify_device_objects.py

```diff
@@ -1,10 +1,50 @@
+import pydoc
+from collections import defaultdict
+from functools import partial
 from typing import Any, List, MutableMapping
 
+import dask
 from dask.utils import Dispatch
 
 from .proxy_object import ProxyObject, asproxy
 
 dispatch = Dispatch(name="proxify_device_objects")
+ignore_types = None
+
+
+def _register_ignore_types():
+    """Lazy register types that shouldn't be proxified
+
+    It reads the config key "jit-unspill-ignore" (DASK_JIT_UNSPILL_IGNORE),
+    which should be a comma seperated list of types to ignore. The default
+    value is:
+        DASK_JIT_UNSPILL_IGNORE="cupy.ndarray"
+
+    Notice, it is not possible to ignore types explicitly handled by this
+    module such as `cudf.DataFrame`, `cudf.Series`, and `cudf.Index`.
+    """
+    global ignore_types
+    if ignore_types is not None:
+        return  # Only register once
+    else:
+        ignore_types = ()
+
+    ignores = dask.config.get("jit-unspill-ignore", "cupy.ndarray")
+    ignores = ignores.split(",")
+
+    toplevels = defaultdict(set)
+    for path in ignores:
+        if path:
+            toplevel = path.split(".", maxsplit=1)[0].strip()
+            toplevels[toplevel].add(path.strip())
+
+    for toplevel, ignores in toplevels.items():
+
+        def f(paths):
+            global ignore_types
+            ignore_types = ignore_types + tuple(pydoc.locate(p) for p in paths)
+
+        dispatch.register_lazy(toplevel, partial(f, ignores))
```

Comment:
This is more of a nitpick question, but why do you need to separate objects into different toplevels? I can't seem to understand why wouldn't just assigning `path` to a `set` suffice. In that case, this code here would be much simpler.

Reply:
It is because `register_lazy` only looks at the top level module. That is, when it detect that `cupy` is imported it calls `f()`. Also notice, each call to `register_lazy` _overwrites_ the previous registered function

Reply:
Makes sense, thanks for clarifying.